pkg/dockerregistry/server/admin.go

```diff
@@ -48,13 +48,9 @@ func (bh *blobHandler) Delete(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	bd, err := storage.RegistryBlobDeleter(bh.Namespace())
-	if err != nil {
-		bh.Errors = append(bh.Errors, errcode.ErrorCodeUnknown.WithDetail(err))
-		return
-	}
+	vacuum := storage.NewVacuum(bh.Context, StorageDriver)
```

Comment:
Can't the `registry.Blobs()` be cast to `distribution.BlobDeleter` in order to ditch the global `StorageDriver`?

By the way, when is the `StorageDriver` initialized? During the first access to a blob? Can we be sure it's initialized before a call to this handler?